django/utils/text.py

```diff
@@ -57,6 +57,13 @@ def wrap(text, width):
     )
     result = []
     for line in text.splitlines(True):
+        wrapped = wrapper.wrap(line)
+        if not wrapped:
+            # Line contained only whitespace and/or newline characters
+            # that wrapper dropped. Add an empty string to the result
+            # to preserve empty lines.
+            result.append("")
+            continue
         result.extend(wrapper.wrap(line))
```

Comment:
I believe that this should be, so we reuse the already calculated `wrapped` value:
```suggestion
        result.extend(wrapped)
```

Reply:
Thanks for the review @nessita !

I did some more manual regression testing, and I found two changes in behavior.

First, like you thought, before the CVE fix, lines with only whitespace were preserved. Now empty lines with only whitespace are dropped. 

Second is what happens to text with only newlines. Like with whitespaces, newlines were preserved before. Now they're dropped. 

I've added tests for both of these edge cases and fixed my proposed fix.